### PR TITLE
Minor memory saving in real H propagation

### PR DESCRIPTION
**Pull request by bhourahine**
Approximately 20% saving in routine memory use (<10% overall as
complex H and Sinv still allocated), with possible marginal
performance improvement.

### CODE REVIEW

File: src/dftbp/timedep/timeprop.F90 [hunk 1]
```diff
@@ -2097,22 +2094,18 @@ subroutine propagateRhoRealH(this, rhoOld, rho, H1, Sinv, step)
...
-    !$omp parallel do private(i,j)
-    do i=1,this%nOrbs
-      do j=1,this%nOrbs
-        rhoOld(i,j) = rhoOld(i,j) + cmplx(0, -step, dp) * (T4R(i,j) + imag * T5R(i,j)) &
-            + cmplx(0, step, dp) * conjg(T4R(j,i) + imag * T5R(j,i))
-      enddo
-    enddo
-    !$omp end parallel do
+    rhoOld(:,:) = rhoOld + cmplx(0, -step, dp) * (T3R + imag * T4R)&
+        & + cmplx(0, step, dp) * transpose(T3R - imag * T4R)
```
Comment (awvwgk) [inline src/dftbp/timedep/timeprop.F90:2108]: If you remove the `parallel do` at least `workshare` this step.

Comment (bhourahine) [reply src/dftbp/timedep/timeprop.F90:2109]: Suspect it would be better to just build with compiler options to use BLAS for matrix intrinsics.

Comment (awvwgk) [reply src/dftbp/timedep/timeprop.F90:2109]: I don't see any BLAS operations in this step, other than maybe `zscale` or `zaxpy`, which as level 1 operations are unlikely to have large impact.

Comment (bhourahine) [reply src/dftbp/timedep/timeprop.F90:2109]: Yep, level 1 only.

Curious how workshare will schedule the four matrix additions (simultaneously?)  if given enough processors.

Comment (awvwgk) [reply src/dftbp/timedep/timeprop.F90:2109]: I guess this is hardly a time consuming step in the computation. You could run the timepropagation with Intel's VTune to see whether this routine appears in the hot spot / threading analysis.

Comment (bhourahine) [reply src/dftbp/timedep/timeprop.F90:2109]: Yep (this routine does get called a lot, and effectively contains the only rank3 operations for fixed nuclei propagation, as overlap inversion would only happen once in that mode). Previous profiling suggested that for DFTB hamiltonians it's the interpolation of parameter tables that is the costly part.

**Review by awvwgk — APPROVED** [review]

**Pull request merged**